packages/desktop/views/onboarding/views/complete-onboarding/views/EnterNameView.svelte

```diff
@@ -16,6 +18,17 @@
 
     function onContinueClick(): void {
         try {
+            if ($onboardingProfile.restoreProfileType === RestoreProfileType.Stronghold) {
+                const MILLISECONDS_TO_EXPIRE_STRONGHOLD_PASSWORD =
+                    DEFAULT_STRONGHOLD_PASSWORD_TIMEOUT_IN_MINUTES * SECONDS_PER_MINUTE * MILLISECONDS_PER_SECOND
+                const isStrongholdPasswordExpired =
+                    new Date().getTime() - $onboardingProfile?.timeStrongholdLastUnlocked?.getTime() >
+                    MILLISECONDS_TO_EXPIRE_STRONGHOLD_PASSWORD
+                if (isStrongholdPasswordExpired) {
+                    handleError({ message: localize('views.settings.strongholdPasswordTimeout.description') })
+                    throw new Error()
+                }
+            }
```

Comment:
We should try avoiding code duplication, this is repeated

Reply:
Also, we should not manage any of this logic here, we dont even know what is the timeout the bindings are using

Reply:
I would first check what is the error (if there is any), and from there work
- use the error
- if there is no error but there is still a crash preventing the user from proceeding, we could use `isStrongholdUnlocked`